lib/libzfs/libzfs_util.c

```diff
@@ -620,8 +620,16 @@ zfs_nicenum_format(uint64_t num, char *buf, size_t buflen,
 
 	double val;
 
-	if (format == ZFS_NICENUM_RAW) {
-		snprintf(buf, buflen, "%llu", (u_longlong_t)num);
+	if (format == ZFS_NICENUM_RAW || format == ZFS_NICENUM_RAWTIME) {
+		if (format == ZFS_NICENUM_RAWTIME && num == 0) {
+			/*
+			 * RAWTIME prints dashes for zeros since a zero latency
+			 * is invalid.
+			 */
+			snprintf(buf, buflen, "-");
+		} else {
+			snprintf(buf, buflen, "%llu", (u_longlong_t)num);
+		}
```

Comment:
What do you think about structuring this as an `else if` block instead, which I personally find more readable.

```c
        if (format == ZFS_NICENUM_RAW) {
                snprintf(buf, buflen, "%llu", (u_longlong_t)num);
                return;
        } else if (format == ZFS_NICENUM_RAWTIME && num > 0) {
                snprintf(buf, buflen, "%llu", (u_longlong_t)num);
                return;
        } else if (format == ZFS_NICENUM_RAWTIME && num == 0) {
                snprintf(buf, buflen, "%s", "-");
                return;
        }
```